buckets/redis/bucket_factory.go

```diff
@@ -92,29 +143,29 @@ func (bf *bucketFactory) Init(cfg *pbconfig.ServiceConfig) {
 		logging.Printf("Re-established Redis connections in %v", time.Since(connStart))
 	}
 
+	bf.script = redis.NewScript(luaScript)
+
 	logging.Printf("Initialized redis.BucketFactory in %v", time.Since(start))
 }
 
 func (bf *bucketFactory) connectToRedisLocked() {
 	// Set up connection to Redis
 	bf.client = redis.NewClient(bf.redisOpts)
 
-	t, err := bf.client.Time().Result()
+	_, err := bf.client.Touch("areYouAlive?").Result()
 	if err != nil {
 		logging.Printf("Cannot connect to Redis. TIME returned %v", err)
 	} else {
-		logging.Printf("Connection established. Time on Redis server: %v", t)
+		logging.Printf("Connection established")
 	}
-
-	bf.scriptSHA = loadScript(bf.client)
 }
 
 func (bf *bucketFactory) reconnectToRedis(oldClient *redis.Client) {
 	bf.Lock()
 	defer bf.Unlock()
 
 	// Always close connections on errors to prevent results leaking.
-	if err := bf.client.Close(); !isRedisClientClosedError(err) {
+	if err := bf.client.Close(); err != nil && !isRedisClientClosedError(err) {
```

Comment:
Why did you add this nil check?

Reply:
`isRedisClientClosedError` can be inlined to `err != nil && err.Error() == "bla"`. Simplifying this with some [boolean algebra](https://www.electronics-tutorials.ws/boolean/bool_6.html):
`err != nil && !(err != nil && err.Error == "bla")` -> de Morgan's law
`err != nil && (err == nil || err.Error != "bla")` -> distributive law
`(err != nil && err == nil) || (err != nil && err.Error != "bla")` -> complement law
`err != nil && err.Error != "bla"`

versus

`!(err != nil && err.Error == "bla")` -> de Morgan's law
`err == nil || err.Error != "bla"`

Basically without this check, it doesn't actually do what it looks like it does.

Reply:
I introduced this bug in an earlier commit, trying to fix a poorly named function

Reply:
I missed the `!` before isRedisClientClosed.. ignore me.